### PR TITLE
[mono][jit] Improve handling of the ldtoken+ltoken+Type::op_Equal opt…

### PR DESCRIPTION
**Pull request by vargaz**
…imization.

Previously, only comparisons with gshared types were handled, but if gsharing is disabled, its possible to run into cases where both ldtoken calls product a concrete type.

### CODE REVIEW

**Comment by vargaz** [comment]
This saves around 150k on a wasm hello world.

File: src/mono/mono/mini/intrinsics.c [hunk 1]
```diff
@@ -767,10 +767,10 @@ get_class_from_ldtoken_ins (MonoInst *ins)
  * their relation (EQ/NE/NONE).
  */
 static CompRelation
-get_rttype_ins_relation (MonoInst *ins1, MonoInst *ins2, gboolean *vtype_constrained_gparam)
+get_rttype_ins_relation (MonoCompile *cfg, MonoInst *ins1, MonoInst *ins2, gboolean *vtype_constrained_gparam)
 {
-	MonoClass *k1 = get_class_from_ldtoken_ins (ins1);
-	MonoClass *k2 = get_class_from_ldtoken_ins (ins2);
+	MonoClass *k1 = get_class_from_ldtoken_ins (cfg, ins1);
```
Comment (lambdageek) [inline src/mono/mono/mini/intrinsics.c:772]: Is this ever going to inflate a gshared type and make the `mono_class_is_gparam (k1)` case fail more often?

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/mini/intrinsics.c (hunk 1 above)
Comment (vargaz) [reply src/mono/mono/mini/intrinsics.c:772]: gshared types are encoded using a different opcode (OP_RTTYPE).

**Review by lewing — APPROVED** [review]

**Pull request merged**